### PR TITLE
New Resource: `azurerm_log_analytics_data_export_rule`

### PR DESCRIPTION
**Pull request by WodansSon**

### CODE REVIEW

File: website/docs/r/virtual_desktop_application_group.html.markdown [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-subcategory: "Desktop Virtualization"
+subcategory: "DesktopVirtualization"
```
Comment (katbyte) [inline website/docs/r/virtual_desktop_application_group.html.markdown:2]: I don't think we should be changing this?

File: website/docs/r/log_analytics_data_export.html.markdown [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+* `name` - (Required) The name of the Log Analytics Data Export Rule. Changing this forces a new Log Analytics Data Export Rule to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Log Analytics Data Export should exist. Changing this forces a new Log Analytics Data Export Rule to be created.
+
+* `workspace_resource_id` - (Required) The resource ID of the workspace. Changing this forces a new Log Analytics Data Export Rule to be created.
+
+* `destination_resource_id` - (Required) The destination resource ID. It should be a storage account, an event hub namespace or an event hub. If the destination is an event hub namespace, an event hub would be created for each table automatically.
+
+* `table_names` - (Required) A list of table names to export to the destination resource, for example: ["Heartbeat", "SecurityEvent"].
```
Comment (katbyte) [inline website/docs/r/log_analytics_data_export.html.markdown:58]: could we quote the values property?
```suggestion
* `table_names` - (Required) A list of table names to export to the destination resource, for example: `["Heartbeat", "SecurityEvent"]`.
```

File: azurerm/internal/services/loganalytics/tests/log_analytics_data_export_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+}
+
+func testAccAzureRMLogAnalyticsDataExportRule_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-lader-%d"
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/tests/log_analytics_data_export_resource_test.go:175]: we usually just put the service here
```suggestion
  name     = "acctestRG-la-%d"
```

but this is ok tbh

File: azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	}
+
+	parameters := operationalinsights.DataExport{
+		DataExportProperties: &operationalinsights.DataExportProperties{
+			Destination: destination,
+			TableNames:  utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List()),
+			Enable:      utils.Bool(d.Get("enabled").(bool)),
+		},
+	}
+
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go:121]: _minor_ this should be
```suggestion
	parameters := operationalinsights.DataExport{
		DataExportProperties: &operationalinsights.DataExportProperties{
			Destination: &operationalinsights.Destination{
				ResourceID: utils.String(d.Get("destination_resource_id").(string)),
			},
			TableNames:  utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List()),
			Enable:      utils.Bool(d.Get("enabled").(bool)),
		},
	}

```

File: azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+		return fmt.Errorf("retrieving Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.ResourceGroup, id.WorkspaceName, err)
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("workspace_resource_id", id.WorkspaceID)
+	if props := resp.DataExportProperties; props != nil {
+		d.Set("export_rule_id", props.DataExportID)
+		d.Set("destination_resource_id", flattenArmDataExportDestination(props.Destination))
+		d.Set("enabled", props.Enable)
+		if tableNames := props.TableNames; tableNames != nil {
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go:165]: i think the utility function checks for this?

**Review by katbyte — APPROVED** [review]
Thanks @WodansSon - aside from a few comments this LGTM 👍

File: website/docs/r/virtual_desktop_application_group.html.markdown (hunk 1 above)
Comment (WodansSon) [reply website/docs/r/virtual_desktop_application_group.html.markdown:2]: Yeah, I got called out by the linter so I changed it to what was in the registration code... I'll switch it back to what it was originally. 😄

File: azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go (hunk 5 above)
Comment (WodansSon) [reply azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go:163]: Yep, just looked... if it's `nil` it returns an empty slice. 🙂

**Comment by WodansSon** [comment]
image

**Pull request merged**